code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,313 @@
+		playsound(get_turf(src), 'sound/weapons/bite.ogg', 50, 1)
+		var/damage = rand(melee_damage_lower, melee_damage_upper)
+		target.adjustBruteLoss(damage)
+		health += (damage/2)
+		if(maxHealth < health_cap)
+			maxHealth += 5 //A well fed zombie is a scary zombie
+		times_eaten += 1
+	stop_automated_movement = 0
+	busy = 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/check_evolve()
+	if(!can_evolve) //How did you get here if not?
+		return
+
+	/*
+					Turned (Just reanimated, can be turned back)
+										V
+				Rotting (If eaten once, or died once, can't be turned back)
+				/													\
+			Putrid													Crimson
+	Eaten too much, died too little								Eaten too little, died too much
+	*/
+/*	if(istype(src, /mob/living/simple_animal/hostile/necro/zombie/turned))
+	else if (istype(src, /mob/living/simple_animal/hostile/necro/zombie/rotting))
+		*/
+
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/evolve(var/mob/living/simple_animal/evolve_to)
+	if(ispath(evolve_to, /mob/living/simple_animal/hostile/necro))
+		var/mob/living/evolution = new evolve_to(src.loc,,)
+		evolution.name = name //We want to keep the name
+		if(mind)
+			mind.transfer_to(evolution) //Just in the offchance we have a player in control
+		qdel(src)
+	else
+		//Now, how did you get here when this is supposed to be the zombie evolution tree?
+		new evolve_to(src.loc)
+		qdel(src)
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/drop_host()
+
+/mob/living/simple_animal/hostile/necro/zombie/delayedRegen()
+	..()
+	times_revived += 1
+
+/mob/living/simple_animal/hostile/necro/zombie/turned //Not very useful
+	icon_state = "zombie_turned" //Looks almost not unlike just a naked guy to potentially catch others off guard
+	icon_living = "zombie_turned"
+	icon_dead = "zombie_turned_dead"
+	desc = "A reanimated corpse that looks like it has seen better days. This one still appears quite human."
+	maxHealth = 50
+	health = 50
+	can_evolve = TRUE
+	var/mob/living/carbon/human/host //Whoever the zombie was previously, kept in a reference to potentially bring back
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/check_evolve()
+	..()
+	if(times_revived > 0 || times_eaten > 0)
+		evolve(/mob/living/simple_animal/hostile/necro/zombie/rotting)
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/drop_host()
+	qdel(host) //Bye bye
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/gib()
+	..()
+	drop_host()
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/dust()
+	..()
+	drop_host()
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/attackby(var/obj/item/weapon/W, var/mob/user)
+	..()
+	if(stat == DEAD) //Can only attempt to unzombify if they're dead
+		if(istype (W, /obj/item/weapon/storage/bible)) //This calls for divine intervention
+			var/obj/item/weapon/storage/bible/bible = W
+			visible_message("\The [user] begins whacking at [src] repeatedly with a bible for some reason.")
```

Comment:
You can change this to user.visible_message and have the below to_chat be the second argument to it